entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/impl/EntityRecordService.java

```diff
@@ -65,26 +72,44 @@ public EntityRecord update (EntityRecord entityRecord){
     /**
      * Creates an {@link EntityRecord} from an {@link EntityPreview}, which
      * is then persisted.
-     * 
-     * @param entityCreationRequest         de-referenced XML response instance from Metis
+     *
+     * @param entityCreationRequest de-referenced XML response instance from Metis
      * @param entityCreationRequest entity request object
-     * @param externalEntityType
-	 * @return Saved Entity record
+     * @param externalEntityType    entity type based on de-referencing response
+     * @return Saved Entity record
      * @throws EntityCreationException if an error occurs
      */
     public EntityRecord createEntityFromRequest(EntityPreview entityCreationRequest, String externalEntityType)
-	    throws EntityCreationException {
-	Entity entity = EntityObjectFactory.createEntityObject(externalEntityType);
+            throws EntityCreationException {
+        // Fail quick if no datasource is configured
+        Optional<DataSource> externalDatasourceOptional = datasources.getDatasource(entityCreationRequest.getId());
+        if (externalDatasourceOptional.isEmpty()) {
+            throw new EntityCreationException("No configured datasource for entity " + entityCreationRequest.getId());
+        }
 
-	entity.setPrefLabelStringMap(entityCreationRequest.getPrefLabel());
-	entity.setAltLabel(entityCreationRequest.getAltLabel());
+        Entity entity = EntityObjectFactory.createEntityObject(externalEntityType);
+        long dbId = entityRecordRepository.generateAutoIncrement(entity.getType());
 
-	// TODO: add proxies and aggregations
+
+        entity.setPrefLabelStringMap(entityCreationRequest.getPrefLabel());
```

Comment:
according to step 9.e. in the specs the empty consolidated version and aggregation is saved. We should not add the prefLabel and altLabels properties to the consolidated version at creation time